src/components/Post/ActionsBar/ReportButton/index.tsx

```diff
@@ -33,7 +41,7 @@ const reasons = [
 ];
 
 const ReportButton: FC<IReportButtonProps> = (props) => {
-	const { type, postId, commentId, replyId, className, proposalType } = props;
+	const { type, postId, commentId, replyId, className, proposalType, isDeleteModal , isReply, isPost, onDeleteReply, onDeleteComment , onDeletePost, allowed_roles } = props;
```

Comment:
`onDeleteReply, onDeleteComment , onDeletePost` can be replaced with `onSuccess`

Reply:
instead of increasing props with `isReply, isPost` use type = 'post | comment | reply`
Please use 

`EReportType {
 POST,
 COMMENT,
 REPLY
}`

Reply:
in types.ts